### PR TITLE
PPTP-2366 - Liability flags to send in-flight registrations to updated section

### PR DESCRIPTION
**Pull request by C-Thomp**

### CODE REVIEW

**Review by ajchilts — COMMENTED** [review]
As everyone will be pushed through liabilities at some stage soon there is a lot of throw away code and spec tests here that will become redundant in time.

My initial approach was to push a user to an explanation page from CYA page load if the flags were not set. Then force the user through Liability until they complete it. Showing the explanation page each time.

This way there would have been no access to CYA until Liability was complete. Therefore no changes to CYA or CYA is-complete logic would have been required. 

I get this maybe looks better from a design perspective but it comes with some degree of tech debt.

**Review by ajchilts — APPROVED** [review]

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/TaskListController.scala [hunk 1]
```diff
@@ -19,39 +19,39 @@ package uk.gov.hmrc.plasticpackagingtax.registration.controllers
...
-        Ok(partnershipPage(request.registration, startLink))
+        Ok(partnershipPage(request.registration, startLink, showRestartLiabilityContent))
       else
-        Ok(singleEntityPage(request.registration, startLink))
+        Ok(singleEntityPage(request.registration, startLink, showRestartLiabilityContent))
     }
 
+  private def hasCompletedOldLiability(request: JourneyRequest[AnyContent]) = {
+    request.registration.liabilityDetails.newLiabilityFinished.isEmpty ||
+      request.registration.liabilityDetails.newLiabilityStarted.isEmpty
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/TaskListController.scala]: So we're saying it's "old" if either of the two "new liability" flags is missing... I think that is correct?

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala [hunk 2]
```diff
@@ -101,10 +99,12 @@ case class Registration(
 
   def isCompanyDetailsComplete: Boolean = companyDetailsStatus == TaskStatus.Completed
 
-  def companyDetailsStatus: TaskStatus =
-    if (!isLiabilityDetailsComplete)
+  def companyDetailsStatus: TaskStatus = {
+    val x = organisationDetails.status
```
[the remark this answers is not in the document]
Comment (lmcewanhmrc) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala:103]: `y` do you ? 😆

File: app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/TaskListController.scala (hunk 1 above)
[the remark this answers is not in the document]
Comment (C-Thomp) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/controllers/TaskListController.scala]: we think the || should account for someone who did have on old lib and then the change over happened while they are mid section

File: app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala (hunk 2 above)
[the remark this answers is not in the document]
Comment (C-Thomp) [reply app/uk/gov/hmrc/plasticpackagingtax/registration/models/registration/Registration.scala:103]: will do 👍

**Pull request merged**